iocore/cache/Cache.cc

```diff
@@ -3366,7 +3366,8 @@ HTTPInfo_v21::copy_and_upgrade_unmarshalled_to_v23(char *&dst, char *&src, size_
   // Extra data is fragment table - set that if we have it.
   if (n_frags) {
     static size_t const IFT_SIZE = HTTPCacheAlt_v23::N_INTEGRAL_FRAG_OFFSETS * sizeof(FragOffset);
-    size_t ift_actual            = min(n_frags, HTTPCacheAlt_v23::N_INTEGRAL_FRAG_OFFSETS) * sizeof(FragOffset);
+    size_t ift_actual            = HTTPCacheAlt_v23::N_INTEGRAL_FRAG_OFFSETS * sizeof(FragOffset);
+    ift_actual                   = std::min(n_frags * sizeof(FragOffset), ift_actual);
```

Comment:
Why multiple 2x now?

Reply:
Because C++ is stupid. For some reason if I put `HTTPCacheAlt_v23::N_INTEGRAL_FRAG_OFFSETS` inside the `std::min` it drops the `_v23` and then complains about an undefined symbol at link time. I tried encapsulating it in `()` but that didn't change anything.

Reply:
> ../iocore/cache/libinkcache.a(Cache.o): In function `cache_bc::HTTPInfo_v21::copy_and_upgrade_unmarshalled_to_v23(char*&, char*&, unsigned long&, int, unsigned long*)':
> /home/phil/git/trafficserver/trafficserver/iocore/cache/Cache.cc:3369: undefined reference to `HTTPCacheAlt::N_INTEGRAL_FRAG_OFFSETS'

Reply:
Works for me :)

Reply:
Two things to do here:

* Change the `const` in the declaration in `HTTP.h` to `constexpr`.
* In `proxy/hdrs/HTTP.cc` near line 1812, add the declaration
```
int constexpr HTTPCacheAlt::N_INTEGRAL_FRAG_OFFSETS;
```